src/data_structures/interval_tree.rs

```diff
@@ -8,32 +8,62 @@ use std::fmt::Debug;
 use std::ops::Range;
 
 #[derive(Debug, Clone)]
-pub struct IntervalTree<N, D> {
+pub struct IntervalTree<N: Ord + Clone + Debug, D> {
     root: Option<Node<N, D>>,
 }
 
-#[derive(PartialEq, Eq, Debug, Clone)]
-pub struct Entry<'a, N: 'a, D: 'a> {
+#[derive(Debug, Clone)]
+pub struct Entry<'a, N: 'a + Ord + Clone + Debug,  D: 'a + Debug> {
     data: &'a D,
-    interval: &'a Range<N>,
+    interval: &'a Interval<N>,
 }
 
-impl<'a, N: 'a, D: 'a> Entry<'a, N, D> {
+impl<'a, N: 'a + Ord + Clone + Debug, D: 'a+ Debug> Entry<'a, N, D> {
     /// Get a reference to the data for this entry
     pub fn data(&self) -> &'a D {
         self.data
     }
 
     /// Get a reference to the interval for this entry
-    pub fn interval(&self) -> &'a Range<N> {
-        self.interval
+    pub fn interval(&self) -> &'a Interval<N> {
+        &self.interval
     }
 }
 
 
-pub struct IntervalTreeIterator<'a, N: 'a, D: 'a> {
+pub struct IntervalTreeIterator<'a, N: 'a + Ord + Debug + Clone, D: 'a> {
     nodes: Vec<&'a Node<N, D>>,
-    interval: Range<N>,
+    interval: Interval<N>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Interval<N: Ord + Clone + Debug>(Range<N>);
```

Comment:
Better use the `newtype_derive` macro (see `bio::stats::prob``). It will allow you to derive Deref for the wrapped Range. By this, the newtype just needs to do a check in the new() method, the rest can be omitted.